docker/docker_official_image_build_test.py

```diff
@@ -0,0 +1,128 @@
+    try:
+        execute(command.split())
+    except:
+        raise SystemError("Docker Image Build failed")
+    finally:
+        shutil.rmtree(temp_dir_path)
+
+
+def run_jvm_tests(image, tag, kafka_url):
+    temp_dir_path = tempfile.mkdtemp()
+    try:
+        current_dir = os.path.dirname(os.path.realpath(__file__))
+        copy_tree(f"{current_dir}/test/fixtures", f"{temp_dir_path}/fixtures")
+        execute(["wget", "-nv", "-O", f"{temp_dir_path}/kafka.tgz", kafka_url])
+        execute(["mkdir", f"{temp_dir_path}/fixtures/kafka"])
+        execute(["tar", "xfz", f"{temp_dir_path}/kafka.tgz", "-C",
+                f"{temp_dir_path}/fixtures/kafka", "--strip-components", "1"])
+        failure_count = run_tests(f"{image}:{tag}", "jvm", temp_dir_path)
+    except:
+        raise SystemError("Failed to run the tests")
+    finally:
+        shutil.rmtree(temp_dir_path)
+    test_report_location_text = f"To view test report please check {current_dir}/test/report_jvm.html"
+    if failure_count != 0:
+        raise SystemError(
+            f"{failure_count} tests have failed. {test_report_location_text}")
+    else:
+        print(f"All tests passed successfully. {test_report_location_text}")
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "image", help="Image name that you want to keep for the Docker image")
+    parser.add_argument("--image-tag", "-tag", default="latest",
+                        dest="tag", help="Image tag that you want to add to the image")
+    parser.add_argument("--image-type", "-type", choices=[
+                        "jvm"], default="jvm", dest="image_type", help="Image type you want to build")
+    parser.add_argument("--kafka-url", "-u", dest="kafka_url",
+                        help="Kafka url to be used to download kafka binary tarball in the docker image")
+    parser.add_argument("--kafka-version", "-v", dest="kafka_version",
+                        help="Kafka version for which the source for docker official image is to be built")
+    parser.add_argument("--build", "-b", action="store_true", dest="build_only",
+                        default=False, help="Only build the image, don't run tests")
+    parser.add_argument("--test", "-t", action="store_true", dest="test_only",
+                        default=False, help="Only run the tests, don't build the image")
+    args = parser.parse_args()
+
+    if args.image_type == "jvm" and (args.build_only or not (args.build_only or args.test_only)):
+        if args.kafka_url and args.kafka_version:
+            build_jvm(args.image, args.tag, args.kafka_version)
+        else:
+            raise ValueError(
+                "--kafka-url and --kafka-version are required argument for jvm docker official image image")
+
+    if args.image_type == "jvm" and (args.test_only or not (args.build_only or args.test_only)):
+        run_jvm_tests(args.image, args.tag, args.kafka_url)
```

Comment:
Do we expect there will soon be other supported image types? If not the argument should be removed, both here and in `docker/prepare_docker_official_image_source.py`.

Reply:
KIP 974 (https://cwiki.apache.org/confluence/display/KAFKA/KIP-974%3A+Docker+Image+for+GraalVM+based+Native+Kafka+Broker) also aims to introduce a Docker Image for GraalVM based Native Kafka Broker. These arguments have been kept in case we decide to release a docker official image for the same in the future.